src/components/Popover/__snapshots__/Popover.test.tsx.snap

```diff
@@ -19,247 +19,251 @@ exports[`<Popover /> BottomLeft story renders snapshot 1`] = `
   <div
     data-focus-lock-disabled="false"
   >
-    <article
-      aria-describedby="popover-description-1"
+    <div
       aria-hidden="false"
-      aria-labelledby="popover-heading-1"
       class="popover eds-is-active popover--bottom-left"
       data-testid="popover-test"
-      role="dialog"
+      role="button"
       tabindex="0"
     >
-      <div
-        class="popover__content"
+      <article
+        aria-describedby="popover-description-1"
+        aria-labelledby="popover-heading-1"
+        role="dialog"
       >
-        <header
-          class="popover__header"
+        <div
+          class="popover__content"
         >
-          <div
-            class="popover__header-content"
+          <header
+            class="popover__header"
           >
-            <h3
-              class="heading heading--size-body-sm"
-              id="popover-heading-1"
-            >
-              Notifications (4)
-            </h3>
-            <span
-              class="popover__header-title-after"
+            <div
+              class="popover__header-content"
             >
-              <button
-                class="clickable-style button button--icon clickable-style--sm clickable-style--icon clickable-style--brand"
-                type="button"
+              <h3
+                class="heading heading--size-body-sm"
+                id="popover-heading-1"
               >
-                Mark All Seen
-              </button>
-            </span>
-          </div>
-        </header>
-        <div
-          class="popover__body"
-        >
-          <ul
-            class="notification-list"
+                Notifications (4)
+              </h3>
+              <span
+                class="popover__header-title-after"
+              >
+                <button
```

Comment:
Here's another case of a button in a button. This one definitely needs the escape key close logic. I'm looking around the web for advice, and I'm starting to think we should just leave the lint suppression comments 😞 I was hoping there was some role that we were missing that would solve our problem, but it might just be that these lint rule were designed to make sure people aren't using `div`s to make buttons (and forgetting to add `role="button"`) but these cases are exceptions. 🤔

Reply:
The lint rules are a tool we should generally follow, but it's totally fine to suppress them when they don't apply to our context.

I'm sure this rule was made to prevent people from applying random click handlers to divs when they should've had button.

Reply:
I have reversed `Popover.tsx`and snapshot back. to just leave the lint suppression comments.